src/pages/SomePage2/SomePage2.tsx

```diff
@@ -0,0 +1,66 @@
+import { useEffect, useState } from "react";
+
+import { useParams } from "react-router-dom";
+
+import ProductsDetailCard from "./components/ProductsCard/ProductsDetailCard";
+import style from "./SomePage2.module.scss";
+import { getProduct } from "../../api/getProduct";
+import { getProducts } from "../../api/getProducts";
+import { Product } from "../../types/api";
+import ProductsCard from "../ProductsPage/components/productsCard/ProductsCard";
+
+const SomePage2 = () => {
+  const { id } = useParams();
+
+  const [product, setProduct] = useState<Product>();
```

Comment:
по логике тут либо есть продукт/либо нет, в таком случае
```suggestion
  const [product, setProduct] = useState<Product | null>(null);
```

Reply:
сделал